### PR TITLE
Add basic unit test for libvmemcache

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vmemcache/8)

### CODE REVIEW

**Comment by ldorau** [comment]
This PR contains PR https://github.com/pmem/vmemcache/pull/4 yet.

File: tests/test-basic-valgrind.cmake [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+include(${SRC_DIR}/helpers.cmake)
+
+setup()
+
+execute(0 ${SRC_DIR}/valgrind-leak-check.sh "${TEST_DIR}/vmemcache_test_basic /tmp/")
```
Comment (marcinslusarz) [inline tests/test-basic-valgrind.cmake]: No shell scripts please, unless it's absolutely necessary.

File: tests/test-basic.cmake [hunk 2]
```diff
@@ -33,6 +33,6 @@ include(${SRC_DIR}/helpers.cmake)
 
 setup()
 
-execute(0 ${TEST_DIR}/example_test)
+execute(0 ${TEST_DIR}/vmemcache_test_basic /tmp/)
```
Comment (marcinslusarz) [inline tests/test-basic.cmake]: No hardcoded (/tmp) paths please. This should be configurable with default value set to build directory.

Comment (ldorau) [reply tests/test-basic.cmake]: Done

File: tests/test-basic-valgrind.cmake (hunk 1 above)
Comment (ldorau) [reply tests/test-basic-valgrind.cmake]: Done

**Comment by ldorau** [comment]
@kilobyte please review

**Comment by ldorau** [comment]
@kilobyte @janekmi This PR and PR https://github.com/pmem/vmemcache/pull/23 are the most important PRs now, because there have to be tests in the repo that can validate any new PRs.

**Pull request merged**